src/celengine/virtualtex.cpp

```diff
@@ -244,147 +318,65 @@ void VirtualTexture::populateTileTree()
     // Count the number of resolution levels present
     unsigned int maxLevel = 0;
 
-    // Crash potential if the tile prefix contains a %, so disallow it
-    string pattern;
-    if (tilePrefix.find('%') == string::npos)
-        pattern = tilePrefix + "%d_%d.";
-
     for (int i = 0; i < MaxResolutionLevels; i++)
     {
         fs::path path = tilePath / fmt::format("level{:d}", i);
         std::error_code ec;
-        if (fs::is_directory(path, ec))
+        if (!fs::is_directory(path, ec))
+            continue;
+
+        maxLevel = i + baseSplit;
+        int uLimit = 2 << maxLevel;
+        int vLimit = 1 << maxLevel;
+
+        for (const auto& d : fs::directory_iterator(path, ec))
         {
-            maxLevel = i + baseSplit;
-            int uLimit = 2 << maxLevel;
-            int vLimit = 1 << maxLevel;
-
-            for (auto& d : fs::directory_iterator(path, ec))
-            {
-                int u = -1, v = -1;
-                if (sscanf(d.path().filename().string().c_str(), pattern.c_str(), &u, &v) == 2)
-                {
-                    if (u >= 0 && v >= 0 && u < uLimit && v < vLimit)
-                    {
-                        // Found a tile, so add it to the quadtree
-                        Tile* tile = new Tile();
-                        addTileToTree(tile, maxLevel, (unsigned int) u, (unsigned int) v);
-                    }
-                }
-            }
+            if (!d.is_regular_file(ec))
```

Comment:
Hopefully no-one is using symlinks in their virtual textures, not sure if we would want to support that.